subsys/bluetooth/mesh/gen_onoff_cli.c

```diff
@@ -119,12 +119,15 @@ int bt_mesh_onoff_cli_set(struct bt_mesh_onoff_cli *cli,
 			  const struct bt_mesh_onoff_set *set,
 			  struct bt_mesh_onoff_status *rsp)
 {
+	if (!set->reuse_transaction) {
+		cli->tid++;
```

Comment:
This will increase tid before using it. Increment should be added after line 130.

Reply:
This is what gen_lvl does, I just made the same behavior here